idaes/power_generation/carbon_capture/mea_solvent_system/property_package/liquid_prop.py

```diff
@@ -0,0 +1,1282 @@
+          #MEA balance
+          yield xa['MEA'] ==  xt['MEA'] + xt['MEA+'] +  xt['MEACOO-']
+          #CO2 balance
+          yield xa['CO2'] ==  xt['CO2'] + xt['HCO3-'] + xt['MEACOO-']
+          #H2O balance
+          yield xa['H2O'] ==  xt['H2O'] + xt['HCO3-']
+          #charge balance
+          yield xt['MEA+'] == xt['HCO3-'] + xt['MEACOO-']
+
+        try:
+            # Try to build speciation model constraints
+            self.speciation_model = ConstraintList(rule=rule_speciation_model)
+        except AttributeError:
+            # If constraint fails, clean up so that DAE can try again later
+            self.del_component(self.conc_comp_true)
+            self.del_component(self.logC)
+            self.del_component(self.speciation_model)
+            raise
+
+    def _make_state_vars(self):
+
+        self.flow_mol = Var(initialize=1.0,
+                            domain=NonNegativeReals,
+                            doc='Total molar flowrate [mol/s]')
+        self.mole_frac = Var(self._params.component_list,
+                             domain=NonNegativeReals,
+                             bounds=(0, 1),
+                      initialize=1 / len(self._params.component_list),
+                      doc='Component mole fractions [-]')
+        self.pressure = Var(initialize=101325,
+                            domain=NonNegativeReals,
+                            doc='Pressure [Pa]')
+        self.temperature = Var(initialize=298.15,
+                               domain=NonNegativeReals,
+                               doc='Temperature [K]')
+
+    def _flow_mol_comp(self):
+        self.flow_mol_comp = Var(self._params.component_list,
+                    initialize=1.0,
+                    domain=NonNegativeReals,
+                    doc='Component molar flowrate [mol/s]')
+
+        def rule_flow_mol_comp(b,j):
+            return  b.flow_mol_comp[j] == b.mole_frac[j]*b.flow_mol
+
+        self.flow_mol_comp_eq = Constraint(self._params.component_list,
+                                rule = rule_flow_mol_comp,
+                                doc ="Component molar flow in liquid phase"
+                                     " [mol/s]" )
+
+    def _mw_ave(self):
+        def rule_mw_ave(blk):
+            return  sum(blk.mole_frac[j]*blk._params.mw[j]
+                       for j in blk._params.component_list)
+
+        self.mw_ave = Expression(rule = rule_mw_ave,
+                            doc ="Average molecular weight of  liquid phase"
+                            "components [kg/mol]" )
+
+    def _mass_frac(self):
+        def rule_mass_frac(blk,j):
+            return  blk.mole_frac[j]*blk._params.mw[j]/blk.mw_ave
+
+        self.mass_frac = Expression(self._params.component_list,
+                            rule = rule_mass_frac,
+                            doc ="weight fraction"
+                            "[-]" )
+
+    def _mass_frac1(self):
```

Comment:
What is "mass_frac1"? I think this needs a different name, and/or a comment explaining what it actually is.

Reply:
The docstring explains what mass_frac1 stands for.

Reply:
Yes, I think we need a different name - maybe `mass_frac_co2_free`? It is longer, but explains clearly what it is.

One other minor thing I saw when reading the doc string to understand - it should be mass fraction, not weight fraction. Mass is measured in kilograms (or similar), whilst weight is measured in Newtons. As a mass fraction is kg/kg, it should not be called a weight fraction (even if we tend to blur the line in common speech).